### PR TITLE
Skip network adapters that are disconnected

### PR DESCRIPTION
**Pull request by ZLJasonG**
Don't fail to initialise if a network adapter is disconnected.

fixes https://github.com/jstasiak/python-zeroconf/issues/326

### CODE REVIEW

**Comment by ZLJasonG** [comment]
#326

File: zeroconf/__init__.py [hunk 1]
```diff
@@ -2290,6 +2290,13 @@ def add_multicast_member(
         elif _errno in err_einval:
             log.info('Interface of %s does not support multicast, ' 'it is expected in WSL', interface)
             return None
+        elif _errno == 10042:
```
Comment (jstasiak) [inline zeroconf/__init__.py:2293]: Is there a constant in the `errno` module that can be used instead of this value?

File: zeroconf/__init__.py [hunk 2]
```diff
@@ -2290,6 +2290,13 @@ def add_multicast_member(
...
+            log.info(
+                'Failed to set socket option, this can happen if '
+                'the network adapter is in a disconnected state',
+                interface
```
Comment (jstasiak) [inline zeroconf/__init__.py:2297]: The `interface` parameter either needs to be removed or the format string needs to include it.

**Review by jstasiak — CHANGES_REQUESTED** [review]
Thank you! Two small comments, looks good otherwise.

File: zeroconf/__init__.py (hunk 1 above)
Comment (ZLJasonG) [reply zeroconf/__init__.py]: Yes I hadn't spotted it, WSAENOPROTOOPT

**Comment by bdraco** [comment]
@ZLJasonG would you please rebase this PR to the latest master?

**Comment by bdraco** [comment]
Fixes #326

**Comment by bdraco** [comment]
Looks like `WSAENOPROTOOPT` is mapped to `ENOPROTOOPT`

https://github.com/python/cpython/blob/main/Modules/errnomodule.c#L253

**Review by bdraco — APPROVED** [review]

**Pull request merged**